server/apollo.go

```diff
@@ -332,13 +363,47 @@ func (apollo *Apollo) HasStrict(permission permissions.Permission) bool {
 	if apollo.User.Admin {
 		return true
 	}
-	if apollo.Organisation == nil {
+	ok, err := apollo.permissions.HasAnyForOrgTree(
+		apollo.Context(),
+		apollo.User.ID,
+		organisation,
+		permission,
+	)
+	if err != nil {
+		slog.Error("Error while checking organisation permissions", "error", err,
+			"organisation_id",
+			organisation,
+		)
 		return false
 	}
+	return ok
+}
+
+// HasInOrganisationStrict returns a boolean indicating whether or not the currently logged in user has the specified permission within the
+// specified organisation (and only there, global permissions and parent organisations are ignored). If no user is logged in, this will always return false.
+func (apollo *Apollo) HasInOrganisationStrict(
+	permission permissions.Permission,
+	organisation core.OrganisationID,
+) bool {
+	if apollo.permissions == nil {
+		slog.Warn(
+			"Trying to use permission system while Apollo does not have access to a permissions.Service!",
+		)
+		return false
+	}
+	if apollo.User == nil {
+		slog.Warn(
+			"Trying to use permission system while no user is logged in!",
+		)
+		return false
+	}
+	if apollo.User.Admin {
+		return true
+	}
```

Comment:
Deze checks komen nu nogal vaak terug. Is het een idee om die te groeperen in een functie?

Reply:
Momenteel 3x denk ik, dat is wat op het randje of dat in een aparte functie moet of niet voor mij, maar gezien de aard van die checks hier (nil checks op dingen die we daarna expliciet gebruiken) zou ik die liever in de functie zelf laten, dat is duidelijker dan hier een generieke andere functie op te roepen vind ik